File: C#/PBTKata/Bank/AccountService.cs
using LanguageExt;
using static LanguageExt.Prelude;

namespace PBTKata.Bank
{
    public static class AccountService
    {
        public static Either<String, Account> Withdraw(Account account, Withdraw command) =>
            HasAlreadyBeenApplied(account, command) ? account : Apply(account, command);

        private static Either<String, Account> Apply(Account account, Withdraw command)
        {
            if (ExceedMaxwithdrawal(account, command))
                return Left($"Amount exceeding your limit of {account.MaxWithdrawal}");
            else if (ExceedBalance(account, command))
                return Left($"Insufficient balance to withdraw : {command.Amount.Value}");
            else return Right(account.Withdraw(command));
        }

        private static bool HasAlreadyBeenApplied(Account account, Withdraw command) => account.Withdraws.Contains(command);
        private static bool ExceedMaxwithdrawal(Account account, Withdraw command) => command.Amount >= account.MaxWithdrawal;
        private static bool ExceedBalance(Account account, Withdraw command) => command.Amount.Value > account.Balance && !account.IsOverdraftAuthorized;
    }
}

File: java/src/main/java/bank/AccountService.java
package bank;

import io.vavr.control.Either;
import lombok.experimental.UtilityClass;

import static io.vavr.control.Either.left;
import static io.vavr.control.Either.right;

@UtilityClass
public class AccountService {
    public static Either<String, Account> withdraw(Account account, Withdraw command) {
        return hasAlreadyBeenApplied(account, command) ?
                right(account)
                : applyWithdraw(account, command);
    }

    private static Either<String, Account> applyWithdraw(Account account, Withdraw command) {
        if (exceedMaxWithdrawal(account, command))
            return left("Amount exceeding your limit of " + account.maxWithdrawal());
        else if (exceedBalance(account, command))
            return left("Insufficient balance to withdraw : " + command.amount().value());
        else return right(account.withdraw(command));
    }

    private static boolean hasAlreadyBeenApplied(Account account, Withdraw command) {
        return account.withdraws().contains(command);
    }

    private static boolean exceedMaxWithdrawal(Account account, Withdraw command) {
        return command.amount().value() >= account.maxWithdrawal();
    }

    private static boolean exceedBalance(Account account, Withdraw command) {
        return command.amount().value() > account.balance() && !account.isOverdraftAuthorized();
    }
}